documentation/src/main/asciidoc/mapstruct-reference-guide.asciidoc

```diff
@@ -594,6 +611,106 @@ You can find the complete example in the
 https://github.com/mapstruct/mapstruct-examples/tree/master/mapstruct-field-mapping[mapstruct-examples-field-mapping]
 project on GitHub.
 
+[[mapping-with-builders]]
+=== Using builders
+
+MapStruct also supports mapping of immutable types via builders.
+When performing a mapping MapStruct checks if there is a builder for the type being mapped.
+This is done via the `BuilderProvider` SPI.
+If a Builder exists for a certain type, than that builder will be used for the mappings.
+
+The default implementation of the `BuilderProvider` requires the following for a type to have a builder:
+
+* The type (`Person`) has a parameterless public static method (builder creation method) that returns a type (`PersonBuilder`) different than the one that needs building
```

Comment:
Split in line + example: The type has a parameterless public static builder creation method that returns a builder of the type. So `Person` has a for example a method `createPerson` that returns `PersonBuilder`.